src/SecureStore.CyberArkCCP/Services/HttpClientManager.cs

```diff
@@ -0,0 +1,61 @@
+﻿using System;
+using System.Net.Http;
+using System.Security.Cryptography.X509Certificates;
+using System.Threading.Tasks;
+using UiPath.Orchestrator.Extensibility.SecureStores;
+
+namespace UiPath.Orchestrator.Extensions.SecureStores.CyberArkCCP.Services
+{
+    internal class HttpClientManager
+    {
+        private readonly HttpClientCache _httpClientCache;
+        private readonly X509CertificateManager _x509CertificateManager;
+        
+        public HttpClientManager(IHttpClientFactory httpClientFactory, X509CertificateManager x509CertificateManager)
+        {
+            _httpClientCache = new HttpClientCache(httpClientFactory);
+            _x509CertificateManager = x509CertificateManager;
+        }
+
+        public async Task<string> GetContentAsync(Uri uri, string clientCertificateThumbprint, string certificateAuthorityThumbprint)
+        {
+            var httpClient = GetHttpClient(clientCertificateThumbprint, certificateAuthorityThumbprint);
+            
+            var response = await httpClient.GetAsync(uri);
+            response.EnsureSuccessStatusCode();
+
+
+            return await response.Content.ReadAsStringAsync();
+        }
+
+        private HttpClient GetHttpClient(string clientCertificateThumbprint, string certificateAuthorityThumbprint)
+        {
+            var clientKey  = (clientCertificateThumbprint ?? string.Empty) + certificateAuthorityThumbprint;
+            X509Certificate2 clientCertificate;
+            try
+            {
+                clientCertificate = _x509CertificateManager.GetPrivateCertificate(clientCertificateThumbprint);
+            }
+            catch (SecureStoreException)
+            {
+                // The certificate does not exist for the specified thumbprint.
+                // It was removed from the certificate store.
+                // Try remove the http client created for the specified thumbprint if any was created.
+                _httpClientCache.TryRemove(clientKey);
+                throw;
+            }
+
+            HttpClient httpClient = null;
+            if (!_httpClientCache.Exists(clientKey))
```

Comment:
You don't need this method. You can just call Get and if it's null, do the add. Otherwise, you're calling `MemoryCache.Get` twice.